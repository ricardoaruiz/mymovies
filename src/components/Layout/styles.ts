import styled from 'styled-components'
import media from 'styled-media-query'

export const MainContent = styled.div`
    ${media.greaterThan('medium')`
        padding-top: 7rem;
    `}
`
