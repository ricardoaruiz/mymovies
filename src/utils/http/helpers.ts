import { NextApiRequest, NextApiResponse } from 'next'

/**
 * Build query params from received request
 * @param req
 * @returns
 */
export const buildParamsFromQuery = (req: NextApiRequest) => {
    const params = Object.keys(req.query).reduce((params, item) => {
        params += `${item}=${req.query[item]}&`
        return params
    }, '')

    return params ? `?${params}` : ''
}

/**
 * Build a error response
 * @param error
 * @param res
 * @returns
 */
export const buildErrorObject = (error: any, res: NextApiResponse) => {
    if (error.response) {
        const { status, statusText } = error.response
        return res.status(status).json({ error: { message: statusText } })
    }

    res.status(500).json({ error: { message: error.message } })
}
