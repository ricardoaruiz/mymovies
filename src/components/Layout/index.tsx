import React from 'react'
import { Footer, Header } from './components'

import * as S from './styles'

const content = [
    {
        title: 'O Básico',
        items: [
            { text: 'Sobre o MyMovies', href: '/' },
            { text: 'Contate-nos', href: '/' },
        ],
    },
    {
        title: 'Envolva-se',
        items: [
            { text: 'Adicionar um novo filme', href: '/' },
            { text: 'Adicionar uma nova série', href: '/' },
        ],
    },
    {
        title: 'Comunidade',
        items: [
            { text: 'Diretrizes', href: '/' },
            { text: 'Discussões', href: '/' },
        ],
    },
    {
        title: 'Legalidade',
        items: [
            { text: 'Termo de uso', href: '/' },
            { text: 'Política de privacidade', href: '/' },
        ],
    },
]

const Layout: React.FC = ({ children }) => {
    return (
        <>
            <Header />

            <S.MainContent>{children}</S.MainContent>

            <Footer logo={{ href: '/' }} content={content} />
        </>
    )
}

export default Layout
