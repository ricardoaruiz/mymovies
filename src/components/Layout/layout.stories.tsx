import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import Layout from '.'

export default {
    title: 'Components/Structural/Layout',
    component: Layout,
} as Meta

export const Default: Story = () => (
    <Layout>
        <h1>Conte√∫do principal</h1>
    </Layout>
)
