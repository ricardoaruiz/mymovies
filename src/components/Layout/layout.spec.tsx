import { renderWithTheme } from 'utils/tests/helpers'

import Layout from '.'

jest.mock('./components/Header', () => ({
    Header: () => <div data-testid="mocked-header">Header</div>,
}))

jest.mock('./components/Footer', () => ({
    Footer: () => <div data-testid="mocked-header">Header</div>,
}))

describe('<Layout />', () => {
    it('should be render correctly', () => {
        const { container } = renderWithTheme(
            <Layout>
                <div data-testid="main-content">Conte√∫do</div>
            </Layout>
        )

        expect(container).toMatchSnapshot()
    })
})
