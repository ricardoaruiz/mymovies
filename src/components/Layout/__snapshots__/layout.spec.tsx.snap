// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should be render correctly 1`] = `
<div>
  <div
    data-testid="mocked-header"
  >
    Header
  </div>
  <div
    class="styles__MainContent-sc-1fdzcc4-0 eQpLcO"
  >
    <div
      data-testid="main-content"
    >
      Conte√∫do
    </div>
  </div>
  <div
    data-testid="mocked-header"
  >
    Header
  </div>
</div>
`
